tensorflow/python/ops/math_ops.py

```diff
@@ -926,9 +928,17 @@ def cast(x, dtype, name=None):
     return x
 
 
+SaturateCastDType = TypeVar("SaturateCastDType",
+                            dtypes.UInt8, dtypes.UInt16, dtypes.UInt32,
```

Comment:
Rather than having to list out all the types, can we have the API support merging groups of types such as complex/real etc?